OSDOCS-21605: Add 4.20.34 z-stream release notes (#118199)
openshift/openshift-docs: modules/zstream-4-20-34.adoc

// Module included in the following assemblies:
//
// * release_notes/ocp-4-20-release-notes.adoc

:_mod-docs-content-type: REFERENCE
[id="zstream-4-20-34_{context}"]
= RHSA-2026:54583 - {product-title} {product-version}.34 bug fix and security update

Issued: 18 August 2026

[role="_abstract"]
{product-title} release {product-version}.34 is now available. The list of fixed issues that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2026:54583[RHSA-2026:54583] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-2026:54580[RHSA-2026:54580] advisory.

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.20.34 --pullspecs
----

[id="zstream-4-20-34-fixed-issues_{context}"]
== Fixed issues

> * Before this update, after `cluster-api-provider-kubevirt` began exposing all virtual machine instance (VMI) interface IP addresses for dual-stack certificate signing request (CSR) approval support, the `NodePool` controller `ClusterNetworkCIDRConflict` condition check flagged OVN-Kubernetes internal overlay IP addresses as conflicting with the cluster network CIDR. As a consequence, all `KubeVirt`-based node pools reported a false positive `ClusterNetworkCIDRConflict` condition after upgrading to multicluster engine for Kubernetes 2.11.0 ({product-title} 4.19), even though the actual machine infrastructure IP addresses did not conflict with the cluster network. With this release, the CIDR conflict detection reports a collision only when all of a machine's non-link-local addresses fall within the cluster network CIDRs. When a machine also has addresses outside the cluster network, the in-network addresses are treated as expected CNI-internal IPs. As a result, `KubeVirt`-based node pools no longer report false positive `ClusterNetworkCIDRConflict` conditions when OVN-Kubernetes overlay IP addresses are present alongside the real infrastructure address.  (link:https://redhat.atlassian.net/browse/OCPBUGS-97923[OCPBUGS-97923])

Reviewer: ```suggestion
* Before this update, after `cluster-api-provider-kubevirt` began exposing all virtual machine instance (VMI) interface IP addresses for dual-stack certificate signing request (CSR) approval support, the `NodePool` controller `ClusterNetworkCIDRConflict` condition check flagged OVN-Kubernetes internal overlay IP addresses as conflicting with the cluster network CIDR. As a consequence, all `KubeVirt`-based node pools reported a false positive `ClusterNetworkCIDRConflict` condition after upgrading to multicluster engine for Kubernetes 2.11.0 ({product-title} 4.19), even though the actual machine infrastructure IP addresses did not conflict with the cluster network. With this release, the CIDR conflict detection reports a collision only when all of a machine's non-link-local addresses fall within the cluster network CIDRs. When a machine also has addresses outside the cluster network, the in-network addresses are treated as expected CNI-internal IPs. As a result, `KubeVirt`-based node pools no longer report false positive `ClusterNetworkCIDRConflict` conditions when OVN-Kubernetes overlay IP addresses are present alongside the real infrastructure address. (link:https://redhat.atlassian.net/browse/OCPBUGS-97923[OCPBUGS-97923])
```

> * Before this update, `Gateway` pods used by layered products such as {rhoai-full} and Red Hat Connectivity Link did not respect the cluster-wide egress proxy configuration when attempting to pull `wasm` plugins. As a consequence, in disconnected environments behind an enterprise HTTP proxy, the `Gateway` pods could not download the required `wasm` plugins. This situation resulted in `HTTP 403` role-based access control (RBAC) errors and failed inference requests. With this release, the Ingress Operator configures the Istio control plane to respect the cluster-wide egress proxy configuration (`proxies.config.openshift.io/cluster`). As a result, `Gateway` pods successfully pull `wasm` plugins in proxied environments without requiring manual configuration.  (link:https://redhat.atlassian.net/browse/OCPBUGS-99476[OCPBUGS-99476])

Reviewer: ```suggestion
* Before this update, `Gateway` pods used by layered products such as {rhoai-full} and Red Hat Connectivity Link did not respect the cluster-wide egress proxy configuration when attempting to pull `wasm` plugins. As a consequence, in disconnected environments behind an enterprise HTTP proxy, the `Gateway` pods could not download the required `wasm` plugins. This situation resulted in `HTTP 403` role-based access control (RBAC) errors and failed inference requests. With this release, the Ingress Operator configures the Istio control plane to respect the cluster-wide egress proxy configuration (`proxies.config.openshift.io/cluster`). As a result, `Gateway` pods successfully pull `wasm` plugins in proxied environments without requiring manual configuration. (link:https://redhat.atlassian.net/browse/OCPBUGS-99476[OCPBUGS-99476])
```

* Before this update, the *Quick Starts* page in the OpenShift web console loaded translations with a stored language code instead of the resolved language of `i18next`, so local variants such as `zh-CN` did not match the `console-app` resource bundle. As a consequence, on the *Quick Starts* page, some strings could appear missing or untranslated when the console language was set to `Chinese (zh-CN)`. With this release, the *Quick Starts* page now applies the `console-app` resource bundle using `i18n.resolvedLanguage`. As a result, *Quick Starts* text displays correctly for `zh-CN` and other local variants. (link:https://redhat.atlassian.net/browse/OCPBUGS-100057[OCPBUGS-100057])

> * Before this update, on a Telecom Boundary Clock (T-BC) Precision Time Protocol (PTP) configuration, the `ts2phc` service could start too early, before the upstream PTP source was stable and the `phc2sys` service was ready. As a consequence, the T-BC took a long time to converge and the `phc2sys` service could set the system clock based on an incorrect `ts2phc` offset. With this release, the `ts2phc` service on a T-BC now waits to start until the upstream source is stable and the `phc2sys` service is ready. As a result, a T-BC does not start the `ts2phc` service against an unqualified upstream source. Note that the Telecom Grandmaster (T-GM) behavior is not changed.(link:https://redhat.atlassian.net/browse/OCPBUGS-105276[OCPBUGS-105276])

Reviewer: ```suggestion
* Before this update, on a Telecom Boundary Clock (T-BC) Precision Time Protocol (PTP) configuration, the `ts2phc` service could start too early, before the upstream PTP source was stable and the `phc2sys` service was ready. As a consequence, the T-BC took a long time to converge and the `phc2sys` service could set the system clock based on an incorrect `ts2phc` offset. With this release, the `ts2phc` service on a T-BC now waits to start until the upstream source is stable and the `phc2sys` service is ready. As a result, a T-BC does not start the `ts2phc` service against an unqualified upstream source. Note that the Telecom Grandmaster (T-GM) behavior is not changed. (link:https://redhat.atlassian.net/browse/OCPBUGS-105276[OCPBUGS-105276])
```

// Module included in the following assemblies:
//
// * release_notes/ocp-4-20-release-notes.adoc

:_mod-docs-content-type: REFERENCE
[id="zstream-4-20-34_{context}"]
= RHSA-2026:54583 - {product-title} {product-version}.34 bug fix and security update

Issued: 18 August 2026

[role="_abstract"]
{product-title} release {product-version}.34 is now available. The list of fixed issues that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2026:54583[RHSA-2026:54583] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-2026:54580[RHSA-2026:54580] advisory.

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.20.34 --pullspecs
----

[id="zstream-4-20-34-fixed-issues_{context}"]
== Fixed issues

> * Before this update, after `cluster-api-provider-kubevirt` began exposing all virtual machine instance (VMI) interface IPs for dual-stack certificate signing request (CSR) approval support, the `NodePool` controller `ClusterNetworkCIDRConflict` condition check flagged OVN-Kubernetes internal overlay IPs, such as the `ovn-k8s-mp0` management port address, as conflicting with the cluster network CIDR. As a consequence, all KubeVirt-based node pools reported a false positive `ClusterNetworkCIDRConflict` condition after upgrading to multicluster engine for Kubernetes 2.11.0 ({product-title} 4.19), even though the actual machine infrastructure IPs did not conflict with the cluster network. With this release, the CIDR conflict detection reports a collision only when all of a machine's non-link-local addresses fall within the cluster network CIDRs. When a machine also has addresses outside the cluster network, the in-network addresses are treated as expected CNI-internal IPs. As a result, KubeVirt-based node pools no longer report false positive `ClusterNetworkCIDRConflict` conditions when OVN-Kubernetes overlay IPs are present alongside the real infrastructure address. (link:https://redhat.atlassian.net/browse/OCPBUGS-97923[OCPBUGS-97923])

Reviewer 2: ```suggestion
* Before this update, after `cluster-api-provider-kubevirt` began exposing all virtual machine instance (VMI) interface IP addresses for dual-stack certificate signing request (CSR) approval support, the `NodePool` controller `ClusterNetworkCIDRConflict` condition check flagged OVN-Kubernetes internal overlay IP addresses as conflicting with the cluster network CIDR. As a consequence, all `KubeVirt`-based node pools reported a false positive `ClusterNetworkCIDRConflict` condition after upgrading to multicluster engine for Kubernetes 2.11.0 ({product-title} 4.19), even though the actual machine infrastructure IP addresses did not conflict with the cluster network. With this release, the CIDR conflict detection reports a collision only when all of a machine's non-link-local addresses fall within the cluster network CIDRs. When a machine also has addresses outside the cluster network, the in-network addresses are treated as expected CNI-internal IPs. As a result, `KubeVirt`-based node pools no longer report false positive `ClusterNetworkCIDRConflict` conditions when OVN-Kubernetes overlay IP addresses are present alongside the real infrastructure address. (link:https://redhat.atlassian.net/browse/OCPBUGS-97923[OCPBUGS-97923])
```
if this bug is for 4.19, why is it even here? anyway, we always use "addresses" with "IP"

> * Before this update, gateway pods used by layered products such as {rhoai-full} and Red Hat Connectivity Link did not respect the cluster-wide egress proxy configuration when attempting to pull `Wasm` plugins. As a consequence, in disconnected environments behind an enterprise HTTP proxy, the Gateway pods could not download the required `Wasm` plugins. This situation resulted in HTTP 403 role-based access control (RBAC) errors and failed inference requests. With this release, the Ingress Operator configures the Istio control plane to respect the cluster-wide egress proxy configuration (`proxies.config.openshift.io/cluster`). As a result, Gateway pods successfully pull `Wasm` plugins in proxied environments without requiring manual configuration. (link:https://redhat.atlassian.net/browse/OCPBUGS-99476[OCPBUGS-99476])

Reviewer 2: ```suggestion
* Before this update, `Gateway` pods used by layered products such as {rhoai-full} and Red Hat Connectivity Link did not respect the cluster-wide egress proxy configuration when attempting to pull `wasm` plugins. As a consequence, in disconnected environments behind an enterprise HTTP proxy, the `Gateway` pods could not download the required `wasm` plugins. This situation resulted in `HTTP 403` role-based access control (RBAC) errors and failed inference requests. With this release, the Ingress Operator configures the Istio control plane to respect the cluster-wide egress proxy configuration (`proxies.config.openshift.io/cluster`). As a result, `Gateway` pods successfully pull `wasm` plugins in proxied environments without requiring manual configuration. (link:https://redhat.atlassian.net/browse/OCPBUGS-99476[OCPBUGS-99476])
```
use `Gateway`  as either an object, or generically as "gateway"; here since it is talking about the object's pods, I went with the object format

> * Before this update, the *Quick Starts* page loaded translations with a stored language code instead of the resolved language of i18next, so locale variants such as `zh-CN` did not match the console-app resource bundle. As a consequence, on the *Quick Starts* page, some strings could appear missing or untranslated when the console language was set to `Chinese (zh-CN)`. With this release, the *Quick Starts* page now look up and apply the console-app resource bundle using `i18n.resolvedLanguage`. As a result, *Quick Starts* text displays correctly for `zh-CN` and other locale variants. (link:https://redhat.atlassian.net/browse/OCPBUGS-100057[OCPBUGS-100057])

Reviewer 2: ```suggestion
* Before this update, the *Quick Starts* page in the OpenShift web console loaded translations with a stored language code instead of the resolved language of `i18next`, so local variants such as `zh-CN` did not match the `console-app` resource bundle. As a consequence, on the *Quick Starts* page, some strings could appear missing or untranslated when the console language was set to `Chinese (zh-CN)`. With this release, the *Quick Starts* page now applies the `console-app` resource bundle using `i18n.resolvedLanguage`. As a result, *Quick Starts* text displays correctly for `zh-CN` and other local variants. (link:https://redhat.atlassian.net/browse/OCPBUGS-100057[OCPBUGS-100057])
```
a lot of mistakes here that a copy edit could fix; please try to catch these before merge review

> * Before this update, on a Telecom Boundary Clock (T-BC) PTP configuration, `ts2phc` could start before the upstream PTP source was stable enough and before `phc2sys` was ready. As a consequence, the T-BC took a long time to converge, and `phc2sys` could adjust the system clock too early based on an irrelevant `ts2phc` offset. With this release, `ts2phc` start on T-BC is delayed until the upstream source is qualified and `phc2sys` is ready; T-GM behavior is unchanged. As a result, T-BC no longer starts `ts2phc` against an unqualified upstream source. (link:https://redhat.atlassian.net/browse/OCPBUGS-105276[OCPBUGS-105276])

Reviewer 2: ```suggestion
* Before this update, on a Telecom Boundary Clock (T-BC) PTP configuration, the `ts2phc` process could start before the upstream PTP source was stable enough and before the `phc2sys` process was ready. As a consequence, the T-BC took a long time to converge, and `phc2sys` could adjust the system clock too early based on an irrelevant `ts2phc` offset. With this release, the `ts2phc` start on T-BC is delayed until the upstream source is qualified and `phc2sys` is ready. As a result, T-BC no longer starts `ts2phc` against an unqualified upstream source. (link:https://redhat.atlassian.net/browse/OCPBUGS-105276[OCPBUGS-105276])
```
what is PTP? (and what is this bug? it is unintelligible)
